express/component_spec.py

```diff
@@ -0,0 +1,328 @@
+"""This module defines classes to represent an Express component specification."""
+import copy
+import json
+import yaml
+import pkgutil
+import types
+import typing as t
+from dataclasses import dataclass, asdict
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+
+from express.exceptions import InvalidComponentSpec
+from express.io_utils import load_yaml
+from express.common import Field
+from express.pipeline_utils import get_kubeflow_type
+
+
+@dataclass
+class Input:
```

Comment:
```suggestion
class InputArgument:
```
Maybe let's call it like that to not confuse it with input subsets